src/core/qgsvectorlayer.cpp

```diff
@@ -4240,6 +4239,13 @@ QgsExpressionContext QgsVectorLayer::createExpressionContext() const
   return QgsExpressionContext( QgsExpressionContextUtils::globalProjectLayerScopes( this ) );
 }
 
+QgsExpressionContext QgsVectorLayer::createExpressionContext( QgsExpressionContext context ) const
+{
+  context.appendScopes( QgsExpressionContextUtils::globalProjectLayerScopes( this ) );
+  return context;
```

Comment:
Allows merging a provided context with the vectorlayer context, otherwise use an empty context as base context to merge on.